src/main/java/org/openstreetmap/atlas/checks/validation/points/AddressPointMatchCheck.java

```diff
@@ -0,0 +1,166 @@
+
+/**
+ * This check identifies Point objects in OSM that have a specified street number (addr:housenumber)
+ * but no specified street name (addr:street) and are not part of an associated street Relation. No
+ * specified street name refers to either having a null value for the street name key, or no street
+ * name key present at all.
+ *
+ * @author savannahostrowski
+ */
+
+public class AddressPointMatchCheck extends BaseCheck
+{
+    private static final long serialVersionUID = 1L;
+    public static final String NO_STREET_NAME_POINT_INSTRUCTIONS = "This node, {0,number,#}, has "
+            + "no street name specified in the address. The street name should likely "
+            + "be one of {1}. These names were derived from nearby nodes.";
+    public static final String NO_STREET_NAME_EDGE_INSTRUCTIONS = "This node, {0,number,#}, has "
+            + "no street name specified in the address. The street name should likely "
+            + "be one of {1}. These names were derived from nearby ways.";
+    public static final String NO_SUGGESTED_NAMES_INSTRUCTIONS = "This node, {0,number,#}, has "
+            + "no street name specified in the address. No suggestions names were found as there were no"
+            + "nearby nodes or ways with street name key tags.";
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList(
+            NO_STREET_NAME_POINT_INSTRUCTIONS, NO_STREET_NAME_EDGE_INSTRUCTIONS,
+            NO_SUGGESTED_NAMES_INSTRUCTIONS);
+    private static final String ADDRESS_STREET_NUMBER_KEY = AddressHousenumberTag.KEY;
+    private static final String POINT_STREET_NAME_KEY = AddressStreetTag.KEY;
+    private static final String EDGE_STREET_NAME_KEY = NameTag.KEY;
+    private static final String STREET_RELATION_ROLE = "street";
+    private static final String ASSOCIATED_STREET_RELATION = "associatedStreet";
+    private static final double BOUNDS_SIZE_DEFAULT = 150.0;
+
+    private final Distance boundsSize;
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    public AddressPointMatchCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.boundsSize = Distance.meters(
+                (Double) configurationValue(configuration, "bounds.size", BOUNDS_SIZE_DEFAULT));
+    }
+
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        // Object is an instance of Point
+        return object instanceof Point
+                // And does not have an Associated Street Relation
+                && !hasAssociatedStreetRelation(object)
+                // And has a street number specified
+                && object.getTag(ADDRESS_STREET_NUMBER_KEY).isPresent()
+                // And if the street name key has a value of null or if the street name key is not
+                // present
+                && Strings.isEmpty(object.tag(POINT_STREET_NAME_KEY));
```

Comment:
Please consider dropping the previous condition (`object.getTag(ADDRESS_STREET_NUMBER_KEY).isPresent()`) and replacing `Strings.isEmpty(object.tag(POINT_STREET_NAME_KEY))` with `Strings.isNullOrEmpty(object.tag(POINT_STREET_NAME_KEY))`

`Strings.isNullOrEmpty(object.tag(POINT_STREET_NAME_KEY))` will return
- true, if tag does not exist
- true, if tag exists, but null
- true, if tag exists, but empty
- false, if tag exists and has non-empty value